### ISSUE TITLE
CPPrestsdk cmake error build failed ubuntu server 18.04

### ISSUE THREAD

**Issue by WoutervDijk** [microsoft/vcpkg#7131]
I tried installing cpprestsdk, but it does not work on ubuntu server 18.04. I have tried multiple fresh installs of vcpkg and different versions of g++. But without success. What I did in this case was as follows: 
`sudo ./bootstrap-vcpkg.sh`
`sudo ./vcpkg integrate install`
`sudo ./vcpkg install boost-asio`
`sudo ./vcpkg install cpprestsdk`

Command line:
> Starting package 20/20: cpprestsdk:x64-linux
> Building package cpprestsdk[core,default-features,websockets]:x64-linux...
> -- Downloading https://github.com/Microsoft/cpprestsdk/archive/v2.10.13.tar.gz...
> -- Extracting source /home/ubuntu/vcpkg/downloads/Microsoft-cpprestsdk-v2.10.13.tar.gz
> -- Using source at /home/ubuntu/vcpkg/buildtrees/cpprestsdk/src/v2.10.13-fe1b687772
> -- Configuring x64-linux-dbg
> -- Configuring x64-linux-rel
> -- Building x64-linux-dbg
> CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
>     Command failed: /home/ubuntu/vcpkg/downloads/tools/cmake-3.14.0-linux/cmake-3.14.0-Linux-x86_64/bin/cmake --build . --config Debug --target install -- -v
>     Working Directory: /home/ubuntu/vcpkg/buildtrees/cpprestsdk/x64-linux-dbg
>     See logs for more information:
>       /home/ubuntu/vcpkg/buildtrees/cpprestsdk/install-x64-linux-dbg-out.log
> 
> Call Stack (most recent call first):
>   scripts/cmake/vcpkg_build_cmake.cmake:96 (vcpkg_execute_build_process)
>   scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
>   ports/cpprestsdk/portfile.cmake:43 (vcpkg_install_cmake)
>   scripts/ports.cmake:73 (include)
> 
> 
> Error: Building package cpprestsdk:x64-linux failed with: BUILD_FAILED
> Please ensure you're using the latest portfiles with `.\vcpkg update`, then
> submit an issue at https://github.com/Microsoft/vcpkg/issues including:
>   Package: cpprestsdk:x64-linux
>   Vcpkg version: 2019.06.21-unknownhash

[install-x64-linux-dbg-out.log](https://github.com/microsoft/vcpkg/files/3350195/install-x64-linux-dbg-out.log)

**Comment by cskrisz** [comment]
This might be the problem:

> `LLVM ERROR: out of memory`

**Comment by WoutervDijk** [comment]
@cskrisz thank you for the suggestion. How would I do that with vcpkg?
Furthermore, I run a T2 instance on AWS. So I only got 1 core to begin with. And the system also only got 1 gb of ram.

**Comment by cskrisz** [comment]
With vcpkg you can set the `VCPKG_MAX_CONCURRENCY` environment variable. It seems the default value is `num_logical_cores + 1`.

**Comment by WoutervDijk** [comment]
@cskrisz where can I find this environment variable in the vcpkg folder? 
I could not find anything about these variables in the documentation unfortunately.

**Comment by cskrisz** [comment]
I found it in the [source](https://github.com/microsoft/vcpkg/blob/master/toolsrc/src/vcpkg/build.cpp#L342). It is an operating system environment variable, in bash, for example, you can do something like this: `VCPKG_MAX_CONCURRENCY=1 ./vcpkg install cpprestsdk`

**Comment by WoutervDijk** [comment]
Progress was made! This time he was able to build the Debug version (so "Building x64-linux-dbg"). However, it stayed stuck at building the release version ("Building x64-linux-rel"). After 1 hour, I interrupted it. I am going to try it now again.

**Comment by WoutervDijk** [comment]
Seems, that after another try, vcpkg can not build the release version. The ui is still responsive, but it just stays stuck at the building unfortunate.

**Comment by cskrisz** [comment]
I found out the concurrency setting is not wired anywhere. But it does work if you change `vcpkg_install_cmake()` to `vcpkg_install_cmake(DISABLE_PARALLEL)` in `ports/cpprestsdk/portfile.cmake`. Still, 3 gigs of RAM was barely enough on desktop Ubuntu, maybe you can set up some swap memory, but then it might take forever to compile.

**Comment by WoutervDijk** [comment]
@cskrisz I can now confirm that the problem is highly likely the lack of RAM. 
I was able to compile the vcpkg package via my WSL ubuntu 18.04 server. Since this was on my pc, it had enough resources (ram / cpu power). 
As a solution, I compiled it their and then copied the vcpkg folder (and .vcpkg folder) to my T2 instance. Where I was then able to use it for my C++ programs (after chmod it and running vcpkg integrate install).
Maybe it would be a nice addition if there is an option in VCPKG to directly download the compiled libraries instead of compiling it on your own. This would be of great addition to smaller systems, like my T2 instance.

Thanks for the help @cskrisz !!
